aziotd/Cargo.toml

```diff
@@ -18,4 +18,5 @@ aziot-certd = { path = "../cert/aziot-certd" }
 aziot-identityd = { path = "../identity/aziot-identityd" }
```

Comment:
You don't need the `toml` dep any more, right?

Reply:
Relevant utility: https://crates.io/crates/cargo-udeps

Reply:
> While compilation of this tool also works on Rust stable, it needs Rust nightly to actually run.

Alas

Reply:
Yeah, while it wouldn't be feasible to integrate it into the CI (it would really bog things down having to install + recompile with nightly), it is a very useful utility to run _locally_ when you're doing these sorts of refactors.

Reply:
CI won't be bogged down. It's just one extra full compile that needs to run in parallel with the 18 other CI tasks that do full compiles (out of 48 total tasks).

The problem is depending on a pinned nightly. We used to do that for cargo-grcov in the iotedge repo and it worked fine initially. Then it got old enough that it could no longer compile our code so it had to be bumped. Then it got old enough that the binaries stopped being hosted on rust-lang.org so rustup couldn't install it any more and had to be bumped. Then...

So we just gave up on the idea.

Reply:
Ah, yes, I forgot that if we were to add it to the CI, it would be as a separate action that isn't tied to the main pipeline (i.e: the big bash file).

I can see how depending on a pinned nightly would be annoying...
I mean, one hack could be to write a shell script that checks that stable version we are using, and then finds the closest nightly release. Just a thought.

Reply:
Unpinned nightly is probably worse than pinned nightly re: reliability.

Reply:
What I'm proposing is a quasi-pinned nightly, that gets automatically re-pinned to whatever nightly is closest to the stable version we are using. It'd be somewhere in the middle of unpinned and pinned.

Reply:
The comment where I mentioned the case of Azure/iotedge was about how periodically updating the pinned nightly was a pain in the ass that nobody wanted to do, and that was just when we absolutely had to because the alternative was a broken build. Your suggestion involves updating the nightly even more often.

Reply:
Right, but I assume the whole reason it didn't work was because it required _manual_ updates. What I'm saying is that we could write the Action where the prereq script _automatically_ tracks the appropriate nightly version without any manual intevention required.

Reply:
Automatically updated nightly is an unpinned nightly.